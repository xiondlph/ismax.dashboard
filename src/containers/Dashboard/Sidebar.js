import React, { Component } from 'react'
import classNames from 'classnames'

import './Sidebar.scss'

export default class Sidebar extends Component {
    render() {
        const { sidebar } = this.props

        return <div className={classNames('dashboard-main-sidebar', `dashboard-main-sidebar_${sidebar}`)}>
            <ul className='dashboard-main-sidebar-menu'>
                <li className='dashboard-main-sidebar-menu-item dashboard-main-sidebar-menu-item_active'>
                    <span className='fa fa-user'></span>
                    <span className='dashboard-main-sidebar-menu-item-text'>Профиль</span>
                </li>
                <li className='dashboard-main-sidebar-menu-item'>
                    <span className='fa fa-credit-card'></span>
                    <span className='dashboard-main-sidebar-menu-item-text'>Оплата</span>
                </li>
                <li className='dashboard-main-sidebar-menu-item'>
                    <span className='fa fa-users'></span>
                    <span className='dashboard-main-sidebar-menu-item-text'>Рефералы</span>
                </li>
                <li className='dashboard-main-sidebar-menu-item'>
                    <span className='fa fa-question'></span>
                    <span className='dashboard-main-sidebar-menu-item-text'>Помощь</span>
                </li>
                <li className='dashboard-main-sidebar-menu-item'>
                    <span className='fa fa-sign-out'></span>
                    <span className='dashboard-main-sidebar-menu-item-text'>Выход</span>
                </li>
            </ul>
        </div>
    }
}