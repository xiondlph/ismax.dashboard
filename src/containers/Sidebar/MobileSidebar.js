/**
 * Мобильная версия контейнера левой боковой навигационной панели
 */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { NavLink } from 'react-router-dom'
import {
    DASHBOARD_SIDEBAR_STATE_COLLAPSED
} from '../../constants/Dashboard'

import './MobileSidebar.scss'

export default class MobileSidebar extends Component {

    constructor(props) {
        super(props)

        this.resize = this.resize.bind(this)
    }

    resize(e) {
        if (e.target.innerWidth <= 600 && this.props.sidebar !== DASHBOARD_SIDEBAR_STATE_COLLAPSED) {
            this.props.setMobileSidebarState(DASHBOARD_SIDEBAR_STATE_COLLAPSED)
        }
    }

    componentDidMount() {
        global.addEventListener('resize', this.resize);
    }

    componentWillUnmount() {
        global.removeEventListener('resize', this.resize);
    }

    collapseMobileSidebarState() {
        this.props.setMobileSidebarState(DASHBOARD_SIDEBAR_STATE_COLLAPSED)
    }

    render() {
        const
            { sidebar } = this.props,
            collapsed = sidebar === DASHBOARD_SIDEBAR_STATE_COLLAPSED

        return (
            <div className={classNames({
                'dashboard-sidebar-mobile': true,
                'dashboard-sidebar-mobile_collapsed': collapsed
            })}>
                <div className='dashboard-sidebar-mobile-mask' onClick={::this.collapseMobileSidebarState}/>
                <div className='dashboard-sidebar-mobile-wrap'>
                    <div className='dashboard-sidebar-mobile-menu'>
                        <div className='dashboard-sidebar-mobile-menu-logo'>
                            <span className='dashboard-sidebar-mobile-menu-logo-icon' />
                            ISMAX icsystem
                        </div>
                        <NavLink
                            to='/'
                            activeClassName='dashboard-sidebar-mobile-menu-item_active'
                            exact
                            className='dashboard-sidebar-mobile-menu-item'
                            onClick={::this.collapseMobileSidebarState}
                        >
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-user' />
                            <div className='dashboard-sidebar-mobile-menu-item-text'>Профиль</div>
                        </NavLink>
                        <NavLink
                            to='/payment'
                            activeClassName='dashboard-sidebar-mobile-menu-item_active'
                            className='dashboard-sidebar-mobile-menu-item'
                            onClick={::this.collapseMobileSidebarState}
                        >
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-credit-card' />
                            <div className='dashboard-sidebar-mobile-menu-item-text'>Оплата</div>
                        </NavLink>
                        <NavLink
                            to='/referrals'
                            activeClassName='dashboard-sidebar-mobile-menu-item_active'
                            className='dashboard-sidebar-mobile-menu-item'
                            onClick={::this.collapseMobileSidebarState}
                        >
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-users' />
                            <div className='dashboard-sidebar-mobile-menu-item-text'>Рефералы</div>
                        </NavLink>
                        <div className='dashboard-sidebar-mobile-menu-item'>
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-question' />
                            <span className='dashboard-sidebar-mobile-menu-item-text'>Помощь</span>
                        </div>
                        <div className='dashboard-sidebar-mobile-menu-item'>
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-sign-out' />
                            <span className='dashboard-sidebar-mobile-menu-item-text'>Выход</span>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

MobileSidebar.propTypes = {
    sidebar: PropTypes.string.isRequired,
    setMobileSidebarState: PropTypes.func.isRequired
}