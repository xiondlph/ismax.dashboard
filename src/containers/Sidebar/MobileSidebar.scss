.dashboard-sidebar-mobile {

  &-wrap {
    background-color: #32404e;
    transition: width 0.5s;
    position: absolute;
    overflow-x: hidden;
    overflow-y: hidden;
    height: 100%;
    z-index: 4;
    width: 0;
    top: 0;

    @include respond-to(mobile) {
      overflow-y: auto;
      width: $sidebar-expanded-width;
    }

    .dashboard-sidebar-mobile_collapsed & {
      overflow-y: hidden;
      width: 0;
    }
  }

  &-mask {
    background-color: #32404e;
    position: fixed;
    display: none;
    height: 100%;
    width: 100%;
    opacity: .3;
    z-index: 3;
    left: 0;
    top: 0;

    @include respond-to(mobile) {
      display: block;
    }

    .dashboard-sidebar-mobile_collapsed & {
      width: 0;
    }
  }

  &-menu {
    padding-top: $sidebar-menu-item-height;
    min-height: calc(100% - 64px);

    &-logo {
      @include box-shadow(0, 1px, 2px, 0, rgba(0, 0, 0, 0.2));
      line-height: $sidebar-menu-item-height;
      height: $sidebar-menu-item-height;
      width: $sidebar-expanded-width;
      background-color: #e05d6f;
      position: absolute;
      font-size: 16px;
      color: #fff;
      z-index: 1;
      top: 0;

      &-icon {
        width: $sidebar-menu-item-height;
        font-family: icsystem;
        display: inline-block;
        text-align: center;
        font-size: 25px;
        float: left;

        &:before {
          content: "\e60c";
        }
      }
    }

    &-item {
      line-height: $sidebar-menu-item-height;
      height: $sidebar-menu-item-height;
      width: $sidebar-expanded-width;
      transition: width 0.5s;
      white-space: nowrap;
      position: relative;
      overflow: hidden;
      font-size: 14px;
      cursor: default;
      display: block;
      color: #adb3b8;

      .dashboard-sidebar_collapsed & {
        width: $sidebar-collapsed-width;

        &:hover {
          overflow: inherit;
        }
      }

      & &-icon {
        line-height: $sidebar-menu-item-height;
        display: inline-block;
        text-align: center;
        margin-left: 5px;
        font-size: 18px;
        width: 59px;
        float: left;
      }

      &-text {
        display: inline-block;
        width: 186px;

        .dashboard-sidebar-menu-item:hover & {
          background-color: #3c4a57;
        }
      }

      &:hover {
        background-color: #3c4a57;
        transition: color 0.5s;
        color: #fff;
      }

      &_active {
        background-color: #475360;

        &:before {
          background-color: #e05d6f;
          position: absolute;
          display: block;
          content: " ";
          height: 100%;
          width: 5px;
          left: 0;
          top: 0;
        }
      }
    }
  }
}