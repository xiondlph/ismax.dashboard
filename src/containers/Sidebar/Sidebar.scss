.dashboard-sidebar {
  background-color: $dashboard-sidebar-gb-color;
  width: $dashboard-sidebar-expanded-width;
  transition: width 0.5s;
  z-index: 1;

  @include respond-to(mobile) {
    position: absolute;
    height: 100%;
    width: 0;
  }

  &_collapsed {
    width: $dashboard-sidebar-collapsed-width;

    @include respond-to(mobile) {
      width: 0;
    }
  }

  &-menu {
    height: 100%;

    &-item {
      line-height: $dashboard-sidebar-menu-item-height;
      color: $dashboard-sidebar-menu-item-text-color;
      height: $dashboard-sidebar-menu-item-height;
      width: $dashboard-sidebar-expanded-width;
      transition: width 0.5s;
      white-space: nowrap;
      position: relative;
      overflow: hidden;
      font-size: 14px;
      cursor: default;
      display: block;

      @include respond-to(mobile) {
        width: 0;
      }

      .dashboard-sidebar_collapsed & {
        width: $dashboard-sidebar-collapsed-width;

        @include respond-to(mobile) {
          width: 0;
        }

        &:hover {
          overflow: inherit;
        }
      }

      & &-icon {
        line-height: $dashboard-sidebar-menu-item-height;
        display: inline-block;
        text-align: center;
        margin-left: 5px;
        font-size: 18px;
        width: 59px;
        float: left;
      }

      &-text {
        display: inline-block;
        width: 186px;

        .dashboard-sidebar-menu-item:hover & {
          background-color: $dashboard-sidebar-menu-item-hover-bg-color;
        }
      }

      &:hover {
        background-color: $dashboard-sidebar-menu-item-hover-bg-color;
        color: $dashboard-text-color-white;
        transition: color 0.5s;
      }

      &_active {
        background-color: $dashboard-sidebar-menu-item-active-bg-color;

        &:before {
          background-color: $dashboard-sidebar-menu-item-marker-bg-color;
          position: absolute;
          display: block;
          content: " ";
          height: 100%;
          width: 5px;
          left: 0;
          top: 0;
        }
      }
    }
  }
}