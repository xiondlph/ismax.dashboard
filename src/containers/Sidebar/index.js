/**
 * Контейнер боковой навигационной панели (слева)
 */

import React, { Component } from 'react'
import classNames from 'classnames'
import { NavLink } from 'react-router-dom'

import './Sidebar.scss'

export default class Sidebar extends Component {
    render() {
        const { sidebar } = this.props

        return (
            <div className={classNames('dashboard-main-sidebar', `dashboard-main-sidebar_${sidebar}`)}>
                <div className='dashboard-main-sidebar-menu'>
                    <NavLink to='/' activeClassName='dashboard-main-sidebar-menu-item_active' exact className='dashboard-main-sidebar-menu-item'>
                        <span className='fa fa-user'></span>
                        <div className='dashboard-main-sidebar-menu-item-text'>Профиль</div>
                    </NavLink>
                    <NavLink to='/payment' activeClassName='dashboard-main-sidebar-menu-item_active' className='dashboard-main-sidebar-menu-item'>
                        <span className='fa fa-credit-card'></span>
                        <div className='dashboard-main-sidebar-menu-item-text'>Оплата</div>
                    </NavLink>
                    <NavLink to='/referrals' activeClassName='dashboard-main-sidebar-menu-item_active' className='dashboard-main-sidebar-menu-item'>
                        <span className='fa fa-users'></span>
                        <div className='dashboard-main-sidebar-menu-item-text'>Рефералы</div>
                    </NavLink>
                    <div className='dashboard-main-sidebar-menu-item'>
                        <span className='fa fa-question'></span>
                        <span className='dashboard-main-sidebar-menu-item-text'>Помощь</span>
                    </div>
                    <div className='dashboard-main-sidebar-menu-item'>
                        <span className='fa fa-sign-out'></span>
                        <span className='dashboard-main-sidebar-menu-item-text'>Выход</span>
                    </div>
                </div>
            </div>
        )
    }
}