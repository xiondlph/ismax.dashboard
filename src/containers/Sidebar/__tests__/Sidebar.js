import React from 'react'
import renderer from 'react-test-renderer'
import { HashRouter } from 'react-router-dom'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Sidebar from '../index'

import {
    DASHBOARD_SIDEBAR_STATE_EXPANDED
} from '../../../constants/Dashboard'

configure({ adapter: new Adapter() })

describe('Sidebar', () => {
    test('Проверка Sidebar снапшота', () => {

        // Делаем активнам пункт меню payment
        window.location.hash = '#/payment'

        const
            sidebar = DASHBOARD_SIDEBAR_STATE_EXPANDED,
            component = renderer.create(
                // Нужно обернуть в Router
                <HashRouter>
                    <Sidebar sidebar={sidebar} />
                </HashRouter>
            )

        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})