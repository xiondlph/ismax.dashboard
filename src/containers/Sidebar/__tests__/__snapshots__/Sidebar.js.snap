// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Проверка Sidebar снапшота 1`] = `
<div
  className="dashboard-sidebar"
>
  <div
    className="dashboard-sidebar-menu"
  >
    <a
      aria-current={false}
      className="dashboard-sidebar-menu-item"
      href="#/"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className="dashboard-sidebar-menu-item-icon fa fa-user"
      />
      <div
        className="dashboard-sidebar-menu-item-text"
      >
        Профиль
      </div>
    </a>
    <a
      aria-current="true"
      className="dashboard-sidebar-menu-item dashboard-sidebar-menu-item_active"
      href="#/payment"
      onClick={[Function]}
      style={Object {}}
    >
      <span
        className="dashboard-sidebar-menu-item-icon fa fa-credit-card"
      />
      <div
        className="dashboard-sidebar-menu-item-text"
      >
        Оплата
      </div>
    </a>
    <a
      aria-current={false}
      className="dashboard-sidebar-menu-item"
      href="#/referrals"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className="dashboard-sidebar-menu-item-icon fa fa-users"
      />
      <div
        className="dashboard-sidebar-menu-item-text"
      >
        Рефералы
      </div>
    </a>
    <div
      className="dashboard-sidebar-menu-item"
    >
      <span
        className="dashboard-sidebar-menu-item-icon fa fa-question"
      />
      <span
        className="dashboard-sidebar-menu-item-text"
      >
        Помощь
      </span>
    </div>
    <div
      className="dashboard-sidebar-menu-item"
    >
      <span
        className="dashboard-sidebar-menu-item-icon fa fa-sign-out"
      />
      <span
        className="dashboard-sidebar-menu-item-text"
      >
        Выход
      </span>
    </div>
  </div>
</div>
`;
